chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +173,219 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
+        """
+        if count == 0:
+            return 0
+
+        rewards = await self.process_blocks(count=count, farm_to=await wallet.get_new_puzzlehash())
+        await self.process_blocks(count=1)
+
+        peak_height = self.full_node.blockchain.get_peak_height()
+        if peak_height is None:
+            raise RuntimeError("Peak height still None after processing at least one block")
+
+        coin_records = await self.full_node.coin_store.get_coins_added_at_height(height=peak_height)
+        block_reward_coins = {record.coin for record in coin_records}
+
+        await wait_for_coins_in_wallet(coins=block_reward_coins, wallet=wallet)
+
+        return rewards
+
+    async def farm_rewards(self, amount: int, wallet: Wallet) -> int:
+        """Farm at least the requested amount of mojos to the passed wallet. Extra
+        mojos will be received based on the block rewards at the present block height.
+        The rewards will be present in the wall before returning.
+
+        Arguments:
+            amount: The minimum number of mojos to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested wallet.
+        """
+        rewards = 0
+
+        if amount == 0:
+            return rewards
+
+        height_before = self.full_node.blockchain.get_peak_height()
+        if height_before is None:
+            height_before = uint32(0)
+
+        for count in itertools.count(1):
+            height = uint32(height_before + count)
+            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+            if rewards >= amount:
+                await self.farm_blocks(count=count, wallet=wallet)
+                return rewards
+
+        raise Exception("internal error")
+
+    async def wait_transaction_records_entered_mempool(self, records: Collection[TransactionRecord]) -> None:
+        """Wait until the transaction records have entered the mempool.  Transaction
+        records with no spend bundle are ignored.
+
+        Arguments:
+            records: The transaction records to wait for.
+        """
+        ids_to_check: Set[bytes32] = set()
+        for record in records:
+            if record.spend_bundle is None:
+                continue
+
+            ids_to_check.add(record.spend_bundle.name())
+
+        while True:
+            found = set()
+            for spend_bundle_name in ids_to_check:
+                tx = self.full_node.mempool_manager.get_spendbundle(spend_bundle_name)
+                if tx is not None:
+                    found.add(spend_bundle_name)
+            ids_to_check = ids_to_check.difference(found)
+
+            if len(ids_to_check) == 0:
+                return
+
+            await asyncio.sleep(0.050)
+
+    async def process_transaction_records(self, records: Collection[TransactionRecord]) -> None:
```

Comment:
it doesn't look like we are actually adding any TX records to the mempool here

Reply:
I think it makes more sense to add them to the mempool first

Reply:
This was intended to make sure that already submitted transactions get processed.  For example, request that the wallet submit the tx, then call this to wait until the tx has been confirmed in a block.  Does it need a new name to help express that? Or do you think that's a bad API to offer?

Reply:
Is there a reason to separate the adding to the mempool and then the farming / waiting?
Worst case, if it's already in the mempool, adding it will have no effect. So it seems like we can avoid some boilerplate by also offering to add to mempool

Reply:
I'm not sure that `push_transaction_and_process(wallet=wallet, node=node, transactions=[tx])` is significantly better than two calls.  For some reason this activity feels like 'a helper' but pushing a transaction sounds like 'normal simple usage of a wallet'.  Burying the latter in the former feels like it might be obfuscation rather than abstraction.  `*shrug*`

Are you expecting that a combined helper would process it through the normal wallet calls or just direct to the node mempool?

Reply:
It should be `push_transaction_and_process(self, records)`

No need to pass in wallet or the node. Just call self.send_transaction before doing the other things

Reply:
i agree we should keep it separate